src/widget/wbeatmenu.cpp

```diff
@@ -10,6 +10,11 @@ WBeatMenu::WBeatMenu(UserSettingsPointer pConfig, QWidget* parent)
         : QMenu(parent), m_pConfig(pConfig) {
 }
 
+void WBeatMenu::popup(const QPoint& pos, QAction* at) {
+    update();
+    QMenu::popup(pos, at);
+}
+
 void WBeatMenu::update() {
```

Comment:
This is easy to confuse with QWidget::update, which is used to redraw widgets. Maybe call this updateFlags?

Reply:
This function does not upload the flags, it updates the menu. So let's call it `updateMenu`